nautilus_trader/adapters/hyperliquid/data.py

```diff
@@ -92,270 +116,356 @@ def __init__(
+        Unsubscribe from instrument updates.
         """
-        Request bars.
-
-        Parameters
-        ----------
-        request : RequestBars
-            The request for bars.
+        self._log.info(f"Unsubscribed from instrument updates for {command.instrument_id}")
 
+    async def _unsubscribe_instruments(self, command: UnsubscribeInstruments) -> None:
         """
-        self._log.warning(f"Bars request not yet implemented for {request.instrument_id}")
+        Unsubscribe from instruments updates.
+        """
+        self._log.info("Unsubscribed from instruments updates")
 
-    # -- TASKS --------------------------------------------------------------------------------------
+    # -- REQUESTS -----------------------------------------------------------------------------------
 
-    async def _connect(self) -> None:
+    async def _request_instrument(self, request: RequestInstrument) -> None:
         """
-        Connect the client.
+        Request instrument definition following standard pattern.
         """
-        self._log.info("Connecting to Hyperliquid...", LogColor.BLUE)
+        instrument = self.instrument_provider.find(request.instrument_id)
+        if instrument:
+            self._handle_data(instrument)
+            self._log.debug(f"Sent instrument {request.instrument_id}")
+        else:
+            self._log.error(f"Instrument not found: {request.instrument_id}")
 
-        # TODO: Implement actual connection logic when HyperliquidHttpClient is available
-        self._log.warning("Hyperliquid connection logic not yet implemented")
+    async def _request_instruments(self, request: RequestInstruments) -> None:
+        """
+        Request multiple instrument definitions following standard pattern.
+        """
+        instruments = []
+        for instrument_id in request.instrument_ids:
+            instrument = self.instrument_provider.find(instrument_id)
+            if instrument:
+                instruments.append(instrument)
+                self._handle_data(instrument)
+                self._log.debug(f"Sent instrument {instrument_id}")
+            else:
+                self._log.warning(f"Instrument not found: {instrument_id}")
 
-        # Simulate connection for now
-        await asyncio.sleep(0.1)
-        self._log.info("Hyperliquid connection established (placeholder)", LogColor.GREEN)
+        if not instruments:
+            self._log.warning("No instruments found for request")
+        else:
+            self._log.info(f"Sent {len(instruments)} instruments")
 
-    async def _disconnect(self) -> None:
+    async def _request_trade_ticks(self, request: RequestTradeTicks) -> None:
         """
-        Disconnect the client.
+        Request historical trade ticks.
         """
-        self._log.info("Disconnecting from Hyperliquid...", LogColor.BLUE)
+        try:
+            trade_ticks = await self._http_client.request_trade_ticks(
```

Comment:
Same as for bars, will be returned as `pyo3_trades`.

Reply:
🙇